src/nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -287,10 +293,11 @@ def run_conversion(
                         chunks=best_gzip_chunk,
                     )
                 image_series_kwargs.update(data=data)
-                if check_regular_series(series=timestamps):
-                    image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
-                else:
-                    image_series_kwargs.update(timestamps=timestamps)
+
+            if check_regular_series(series=timestamps):
+                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
+            else:
+                image_series_kwargs.update(timestamps=timestamps)
```

Comment:
This is great, you read my mind!

Although... I wonder if maybe also we should add a check that internally recorded `fps` value is equal to `timestamps[1] - timestamps[0]` after knowing that the `timestamps` are regular? If unequal, perhaps warn the user and use the timestamps rate since they thought ahead enough to specify that separately?

Reply:
Thanks! I added a new check, and modified this part accordingly. I also added a new unittest to test this behavior, let me know if you think it's unnecessary or you were thinking about something different!

Reply:
Something I particularly love to see when adding warnings is usage of the `self.assertWarnsWith` and `self.assertRaisesWith` functions of the `TestCase` class imported from `hdmf.testing`. This lets you check the exact text that is expected in the raising condition.